vector/v.info/print.c

```diff
@@ -215,13 +215,58 @@ void print_columns(struct Map_info *Map, const char *input_opt,
     if (db_describe_table(driver, &table_name, &table) != DB_OK)
         G_fatal_error(_("Unable to describe table <%s>"), fi->table);
 
+    JSON_Value *root_value = json_value_init_object();
+    JSON_Object *root_object = json_object(root_value);
+
+    JSON_Value *columns_value = json_value_init_array();
+    JSON_Array *columns_array = json_array(columns_value);
+
+    if (format == JSON) {
+        json_object_set_value(root_object, "columns", columns_value);
+    }
+
     ncols = db_get_table_number_of_columns(table);
-    for (col = 0; col < ncols; col++)
-        fprintf(stdout, "%s|%s\n",
-                db_sqltype_name(
-                    db_get_column_sqltype(db_get_table_column(table, col))),
+    for (col = 0; col < ncols; col++) {
+        if (format == JSON) {
+
+            JSON_Value *column_value = json_value_init_object();
+            JSON_Object *column_object = json_object(column_value);
+
+            json_object_set_string(
+                column_object, "name",
                 db_get_column_name(db_get_table_column(table, col)));
 
+            int sql_type =
+                db_get_column_sqltype(db_get_table_column(table, col));
+            json_object_set_string(column_object, "sql_type",
+                                   db_sqltype_name(sql_type));
+
+            int c_type = db_sqltype_to_Ctype(sql_type);
+            json_object_set_boolean(
+                column_object, "is_number",
+                (c_type == DB_C_TYPE_INT || c_type == DB_C_TYPE_DOUBLE));
+
+            json_array_append_value(columns_array, column_value);
+        }
+        else {
+            fprintf(stdout, "%s|%s\n",
+                    db_sqltype_name(
+                        db_get_column_sqltype(db_get_table_column(table, col))),
+                    db_get_column_name(db_get_table_column(table, col)));
+        }
+    }
+
+    if (format == JSON) {
+        char *serialized_string = NULL;
+        serialized_string = json_serialize_to_string_pretty(root_value);
+        if (serialized_string == NULL) {
+            G_fatal_error(_("Failed to initialize pretty JSON string."));
```

Comment:
Looking at #4533 and #4507, I wonder if more careful resource handling is needed in these cases. It is G_fatal_error, so basically exit for v.info, so I'm fine with merging this as is, but perhaps we want to set up some best practices here. @nilason and @aajit232, can you please comment? Feel free to add your 2 cents, @NishantBansal2003.

Reply:
I agree with your suggestion for improved resource handling. I think we should free the resources to prevent any potential leaks. If you'd like, I can modify the code to free resources upon encountering a `G_fatal_error`.

Reply:
Sure, sounds good!

Reply:
In general I'd say it is good to closing database connections and similar (?!) before exit. Otherwise (strings etc.) not necessary.

Reply:
> and similar (?!)

which includes closing open files.

Reply:
Reading my above comments, it was perhaps not clearly expressed. In the case of `G_fatal_error()` it is definitely better to let the system free any in-memory objects like strings, structs etc. but close any file or db connections.  In this particular case only  `db_close_database_shutdown_driver()` and `Vect_close()` should be called.

Freeing in-memory objects before an `exit(EXIT_FAILURE)` which is the result of a call to G_fatal_error, only leads to unnecessary complicated code. (And competent static analysers do not complain on these cases either).

Another problem, if we take the following as an example:
```c
    const char *driver_name = fi->driver;
    Vect_destroy_field_info(fi);
    G_fatal_error(_("Unable to open driver <%s>"), driver_name);
```

the string `driver_name` points to will be freed by `Vect_destroy_field_info` causing the inclusion in the G_fatal_error message to fail.

Reply:
Got it, done...